frontend/src/actionCreators/parcelsActionCreator.tsx

```diff
@@ -8,15 +8,15 @@ import CustomAxios from "customAxios";
 import { AxiosResponse } from "axios";
 import { createRequestHeader } from "utils/RequestHeaders";
 
-export const fetchParcels = (parcelBounds: API.IParcelListParams | null) => (dispatch:Function) => {
+export const fetchParcels = (parcelBounds: API.IParcelListParams | null) => (dispatch: Function) => {
```

Comment:
Is there a reason we're returning a function (callback) in this function?  Seems like a bad design and non-intuitive.

Reply:
This is some redux boilerplate which may look confusing. The title of the file helps to explain a bit, this is an just an action creator, which creates an action for future dispatch using the redux dispatch method. This is essentially just a shorthand way of creating an action.
https://redux.js.org/basics/actions/#action-creators
What you are talking about is a bound action creator, which we could easily wrap our action creators with if we wanted (see the same article as above).

Reply:
For reference, this is how Redux "works" by designing "action creators" ie functions that return a function or also called HOFs (higher-order functions). I agree with you that it is increasingly complex to follow but that's the pattern with Redux unfortunately.

https://redux.js.org/basics/actions/#action-creators

Reply:
Meh, I'm fine with it.  I just am not a fan of **callback hell**.  Any time I see it I wonder why it would be preferred over promises or async - function(...).then(...)

Reply:
No, its not a callback in the way you are thinking. Calling dispatch on the created action returns a promise.